### PR TITLE
Add error if unexpected keys are found on Habitat_Sim configs

### PR DESCRIPTION
**Pull request by Skylion007**
## Motivation and Context

* This breaks out a change from the orthocamera PR that errors Habitat_Sim when an unexpected config is encountered. This also will help ensure proper version / API compatibility between Habitat_Sim and Habitat_Lab.

## How Has This Been Tested

* Locally with pytest 
## Types of changes

- Bug fix (non-breaking change which fixes an issue)

## Checklist

- [X] My code follows the code style of this project.
- [X] I have read the **CONTRIBUTING** document.
- [X] I have completed my CLA (see **CONTRIBUTING**)
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

This may break some old custom sensors that exist as third parties. We can make it less strict if needed or add a way to extend the ignore_keys more easily (perhaps through the config itself).

### CODE REVIEW

File: habitat/sims/habitat_simulator/habitat_simulator.py [hunk 1]
```diff
@@ -61,9 +64,21 @@ def if_config_to_lower(config):
...
+        if has_ignores or low_attr not in ignore_keys:
+            if hasattr(config_to, low_attr):
+                setattr(config_to, low_attr, if_config_to_lower(value))
+            else:
+                # TODO consider new type of Error for this?
+                raise RuntimeError(
+                    f"""{low_attr} is not found on habitat_sim but is found on habitat_lab config.
+                    It's also not in the list of keys to ignore: {ignore_keys}
+                    Did you make a typo in the YAML config?
+                    If not the version of habitat_sim may not be supported: {config_from}
```
Comment (mathfac) [inline habitat/sims/habitat_simulator/habitat_simulator.py:79]:
```suggestion
                    If not the version of Habitat Sim may not be compatible with Habitat Lab version: {config_from}
```

File: habitat/sims/habitat_simulator/habitat_simulator.py [hunk 2]
```diff
@@ -61,9 +64,21 @@ def if_config_to_lower(config):
...
     for attr, value in config_from.items():
-        if hasattr(config_to, attr.lower()):
-            setattr(config_to, attr.lower(), if_config_to_lower(value))
+        low_attr = attr.lower()
+        if has_ignores or low_attr not in ignore_keys:
+            if hasattr(config_to, low_attr):
+                setattr(config_to, low_attr, if_config_to_lower(value))
+            else:
+                # TODO consider new type of Error for this?
+                raise RuntimeError(
```
Comment (mathfac) [inline habitat/sims/habitat_simulator/habitat_simulator.py:75]: Let's solve this todo now: NameError should fit better here.

File: habitat/sims/habitat_simulator/habitat_simulator.py [hunk 3]
```diff
@@ -61,9 +64,21 @@ def if_config_to_lower(config):
...
+                    f"""{low_attr} is not found on habitat_sim but is found on habitat_lab config.
+                    It's also not in the list of keys to ignore: {ignore_keys}
+                    Did you make a typo in the YAML config?
```
Comment (mathfac) [inline habitat/sims/habitat_simulator/habitat_simulator.py:78]: Source of the config can be different.
```suggestion
                    Did you make a typo in the config?
```

File: habitat/sims/habitat_simulator/habitat_simulator.py [hunk 4]
```diff
@@ -62,8 +65,18 @@ def if_config_to_lower(config):
...
+        low_attr = attr.lower()
+        if ignore_keys is None or low_attr not in ignore_keys:
+            if hasattr(config_to, low_attr):
+                setattr(config_to, low_attr, if_config_to_lower(value))
+            else:
+                raise NameError(
+                    f"""{low_attr} is not found on habitat_sim but is found on habitat_lab config.
+                    It's also not in the list of keys to ignore: {ignore_keys}
+                    Did you make a typo in the config?
+                    If not the version of Habitat Sim may not be compatible with Habitat Lab version: {config_from}: {config_from}
```
Comment (mathfac) [inline habitat/sims/habitat_simulator/habitat_simulator.py:77]:
```suggestion
                    If not the version of Habitat Sim may not be compatible with Habitat Lab version: {config_from}.
```

File: habitat/sims/habitat_simulator/habitat_simulator.py [hunk 5]
```diff
@@ -225,18 +238,39 @@ def create_sim_config(
         overwrite_config(
             config_from=self.habitat_config.HABITAT_SIM_V0,
             config_to=sim_config,
+            ignore_keys={"gpu_gpu"},
```
Comment (mathfac) [inline habitat/sims/habitat_simulator/habitat_simulator.py:241]: Can we add some comments to the keys like : hablab only field /  deprecated field or etc.

**Review by mathfac — APPROVED** [review]
Thank you, approved with comments.

File: habitat/sims/habitat_simulator/habitat_simulator.py (hunk 5 above)
Comment (Skylion007) [reply habitat/sims/habitat_simulator/habitat_simulator.py:241]: Where though? In the default? In all the configs?

Comment (erikwijmans) [reply habitat/sims/habitat_simulator/habitat_simulator.py:241]: Probably just here.  A comment at to why we are ignoring that key

File: habitat/sims/habitat_simulator/habitat_simulator.py (hunk 4 above)
Comment (mathfac) [inline habitat/sims/habitat_simulator/habitat_simulator.py:77]: is  doubling config_from `{config_from}: {config_from}` expected?

**Review by erikwijmans — APPROVED** [review]

**Pull request merged**